### PR TITLE
Problem: (CRO-59) No CLI for interacting with client backend

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Added a CLI layer on client backend

### CODE REVIEW

**Comment by devashishdxt** [comment]
> The `ask_passphrase` returns String
> 
> https://github.com/crypto-com/chain/blob/af98d5f993d084804d0a6bfee6ef4566287d90bc/signer-cli/src/commands/address_command.rs#L54
> 
> -- would that get cleared by the wallet client?

This doesn't use `ask_passphrase` from `signer-cli`. Here's the `ask_passphrase` definition in `main.rs`: https://github.com/crypto-com/chain/pull/73/files#diff-22eda7059b43224de055e8390df378ddR56

File: client-cli/src/command/transaction_command.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+use chain_core::tx::data::address::ExtendedAddr;
+use chain_core::tx::data::attribute::TxAttributes;
+use chain_core::tx::data::output::TxOut;
+use client_common::{ErrorKind, Result};
+use client_core::WalletClient;
+
+use crate::ask_passphrase;
+
+#[derive(Debug, StructOpt)]
+pub enum TransactionCommand {
```
Comment (tomtau) [inline client-cli/src/command/transaction_command.rs]: It's almost copy-pasted from https://github.com/crypto-com/chain/blob/master/signer-cli/src/commands/transaction_command.rs

Is there any use case for signer-* ? Would it make sense to remove signer-* and move that functionality into client-* crates (if any)?

**Comment by devashishdxt** [comment]
> Some of the commands are almost identical / duplicate to what's in `signer-*` -- if it makes sense, signer-* could be removed and the functionality moved to client... e.g. if I see it correctly, the only difference between client's transaction command and signer's transaction command is that the client code broadcasts the transaction at the end, whereas the signer just prints it out in the command line.
> I think this difference could just be captured in some special debugging/development flag... something like if "print_only" (default false) is set, it won't broadcast the transaction.
> 
> For address command, I guess the only difference is that the client one is tied to a "wallet"?

It would make sense to remove `signer-*` once we have all the functionalities of `signer` in `client`. Currently, `client` only supports simple transactions whereas `signer` also has an infant version of multisig support. So, I'd vote against removing `signer` until we have multisig support in `client`.

File: client-cli/src/command/transaction_command.rs (hunk 1 above)
Comment (devashishdxt) [reply client-cli/src/command/transaction_command.rs]: `signer-*` should only be removed once we have multisig support in `client`.

**Comment by tomtau** [comment]
@devashishdxt Well, the functionality of `signer` can't be really used at the moment easily, because it doesn't display multisig addresses, so while the signing is present, one won't know what to put in the output.
I've opened CRO-111 epic to track this -- feel free to break it down into more issues

**Pull request merged**